Misc - Fix Switch and Checkbox related bugs

tmcconechy opened the pull request:
**Explain the details for making this change. What existing problem does the pull request solve?**

This PR fixes a bunch of bugs in switch and checkbox. Namely:

- moved checkbox group inside checkbox examples
- added converted tests for both
- fixed `value` vs `checked`

**Related github/jira issue (required)**:
Fixes #1893
Fixes #2021
Fixes #2020
Fixes #2045
Fixes #2027
Fixes #1978

**Steps necessary to review your pull request (required)**:
- checks all pass
- also try testing the form examples in Angular for checkbox and switch
- go to http://localhost:4300/ids-checkbox/required-state.html and resize the page. the asterisk is closer but because of wrapping is outside the container, so may not be 100% perfect with the text but better
- go to http://localhost:4300/ids-checkbox/checkbox-group.html check some checkboxes and call `document.querySelector('ids-checkbox-group').selectedCheckboxes` it should now show whats selected (note renamed)
- go to http://localhost:4300/ids-checkbox/checkbox-group.html check some checkboxes and call `document.querySelector('ids-checkbox-group').value = true` or `document.querySelector('ids-checkbox-group').value = [true, true, true]` and it should update. Check the Dom and either will see a `checked` attribute or no attribute
- go to http://localhost:4300/ids-checkbox/example.html and call `document.querySelector('ids-checkbox').checked = true` `document.querySelector('ids-checkbox').value` should return on then setting `document.querySelector('ids-checkbox').value='test'` if checked should return test in `document.querySelector('ids-checkbox').value`
- go to http://localhost:4300/ids-switch/example.html and call `document.querySelector('ids-switch').checked = true` `document.querySelector('ids-switch').value` should return on then setting `document.querySelector('ids-switch').value='test'` if checked should return test in `document.querySelector('ids-switch').value`

**Included in this Pull Request**:
- [x] Some documentation for the feature.
- [x] A test for the bug or feature.
- [x] A note to the change log.

jdilag90 commented:
Go to http://localhost:4300/ids-checkbox/checkbox-group.html
Run `$('ids-checkbox').value = false;`

![image](https://github.com/infor-design/enterprise-wc/assets/98347152/814dd0d5-ab9d-4071-b637-4dc822dba121)

tmcconechy commented:
@jdilag90 I tried that above and did not see it? Can you try a git pull and rebuild because i know i saw that error at one point and fixed it (in that case value will be ignored since its not a string)

Fixed tests
